addonscript/script.py

```diff
@@ -0,0 +1,73 @@
+import aiohttp
+import asyncio
+import os
+
+import scriptworker.client
+
+from addonscript.api import do_upload, get_upload_status, get_signed_xpi
+from addonscript.task import build_filelist
+from addonscript.xpi import get_langpack_info
+
+
+def _craft_aiohttp_connector(context):
+    return aiohttp.TCPConnector()
+
+
+def get_default_config(base_dir=None):
+    base_dir = base_dir or os.path.dirname(os.getcwd())
+    default_config = {
+        'work_dir': os.path.join(base_dir, 'work_dir'),
+        'schema_file': os.path.join(os.path.dirname(__file__), 'data', 'addonscript_task_schema.json'),
+    }
+    return default_config
+
+
+async def sign_addon(context, locale):
+    # Retry Me
+    if 1:
+        upload_data = await do_upload(context, locale)
+        pk = upload_data['pk']
+    # Retry me, instead of sleeping
+    if 1:
+        await asyncio.sleep(60*6)  # wait 6 minutes
+        upload_status = await get_upload_status(context, locale, pk)
+        if len(upload_status['files']) != 1:
+            raise Exception("Expected 1 file")
+        if (upload_status.get('validation_results') and
+                upload_status['validation_results']['errors']):
+            raise Exception("Automated Validation produced errors")
+        if not upload_status['files'][0]['signed']:
+            raise Exception("expected signed xpi")
+    # Retry me
+    if 1:
+        destination = 'foo'  # XXX: Fixme
+        await get_signed_xpi(context, upload_status['files'][0]['download_url'],
+                             destination)
```

Comment:
We may want a check on the xpi here or in the `get_signed_xpi` function. Valid .xpi is a good start; whether it's signed appropriately is another.